### PR TITLE
Move recursive behavior behind flag option

### LINKED ISSUE #3: Split into recursive and non-recursive functions
This is a good utility library, but the recursion-by-default is surprising. Would you consider releasing a 2.0 with separate functions, one shallow and one recursive?

### PR DESCRIPTION
**Pull request by gilbert**
Let me know what you think.

Closes #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,23 +1,39 @@
 'use strict'
 
-module.exports = function objectifyArray (array, keyName) {
+module.exports = function objectifyArray (array, keys, options) {
```
Comment (zeke) [inline index.js]: What about making `keyNames` an option, with a default value of `['id']`?

**Comment by zeke** [comment]
Not worried about backwards compatibility. We can just publish a new major version with the `(array[, options])` method signature.

If we wanted to help avoid any breakage for users though, we could detect if the second argument is an array and use it to assemble the options object.

File: index.js (hunk 1 above)
Comment (gilbert) [reply index.js]: I had originally coded it to look like `objectify(x, { by: 'id' })`, but it felt a bit verbose as I was revising the examples in the readme. That usually doesn't bother me, but that plus library precedence was enough to push back to what you see in the PR.

**Comment by gilbert** [comment]
A new major version sounds best. It's a breaking change anyway since it is no longer recursive by default.

File: README.md [hunk 2]
```diff
@@ -38,24 +38,52 @@ This will return an object, using the `id` property values as keys:
...
 
 The module exports a single function:
 
-#### `objectifyArray(array[, keyName])`
+#### `objectifyArray(array[, options])`
 
-* `array` Array
-* `keyName` String (optional) - A property other than `id` or `name` to use for the object keys.
+* `array` Array - The array of objects to index.
+* `options` Object (optional) - `{ by: String | Array<String>, recursive: Boolean }`
```
Comment (zeke) [inline README.md]: This looks good to me. Can you enumerate the options in a nested unordered list, and also document the default values for each option?

Comment (gilbert) [reply README.md]: 👍

**Comment by zeke** [comment]
```
~/git/zeke/objectify-array master
$ np major
 ✔ Prerequisite check
 ✔ Git
 ✔ Cleanup
 ✔ Installing dependencies using npm
 ✔ Running tests
 ✔ Bumping version using npm
 ✔ Publishing package
 ✔ Pushing tags

 objectify-array 2.0.0 published 🎉
```

**Pull request merged**